sale_coupon_advanced/models/sale_order_line.py

```diff
@@ -4,8 +4,37 @@
 
 
 class SaleOrderLine(models.Model):
+    """Extend to add forced_reward_line field and override unlink."""
+
     _inherit = "sale.order.line"
 
     forced_reward_line = fields.Boolean(
         help="Tech field to cleanup automatically created lines"
     )
+
+    def unlink(self):
+        """Override to unlink disc lines when reward ones are unlinked.
+
+        Also removes related sale.coupon.program records.
+        """
+        lines_to_remove = self.env['sale.order.line']
+        SaleCouponProgram = self.env[
+            'sale.coupon.program']
+        for line in self:
+            program = SaleCouponProgram.search([
```

Comment:
```suggestion
            programs = SaleCouponProgram.search([
```
there can be a lot of them

Reply:
Right, but then I need to modify logic too. I somehow thought there would be only one. I need to be aware of that when filtering.